### PR TITLE
Bug 1872709: Copy foreign layers from external sources during mirroring

### PR DESCRIPTION
**Pull request by smarterclayton**
A foreign image layer has a URL that points outside the current registry, such that the registry returns not found if the blob
is requested. Since mirror attempts to ensure the destination image can be pulled, even in the presence of disjoint networks,
and given that the Docker daemon will by default attempt the blob lookup first before checking URLs, always copy the blob
by checking the source registry, then trying to pull the layer URLs.

There may be scenarios where this is not desirable, and a future change may allow skipping mirror of those blobs.

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/retest

**Comment by smarterclayton** [comment]
/hold

Testing out whether i can do a silghtly better fix

**Comment by smarterclayton** [comment]
/hold cancel

This is updated to actually mirror URLs after reading the spec and discussing with oleg.  Verified this works between gcr -> internal, internal -> internal, and gcr -> quay.

**Review by soltysh — APPROVED** [review]
Got :+1: from Oleg on slack
/lgtm
/retest

**Comment by dmage** [comment]
title needs to be updated, otherwise LGTM

**Pull request merged**
